connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java

```diff
@@ -468,9 +469,23 @@ public void run() {
                     }
                 });
 
-                if (entityWriter.getType() != NettyEntityWriter.Type.DELAYED) {
-                    entityWriter.flush();
+                headersSet.await();
```

Comment:
As we have to wait anyway, do we really need the executorService?. Maybe we can get rid of it and also the CountDownLatch.

Reply:
Ignore it, the countdownlatch is different.